tools/misc/mapping_extractor.py

```
@@ -0,0 +1,259 @@
+    
+    # Validation
+    if not isinstance(target_column_names, list) or not target_column_names:
+        raise ValueError("\"target_column_names\" must be a non-empty list.")
+    if rows_to_ignore is not None and not isinstance(rows_to_ignore, list):
+        raise ValueError("\"rows_to_ignore\" must be a list if provided.")
+
+    # Try to read the source Excel sheet
+    try:
+        df = pd.read_excel(source_file, sheet_name=source_sheet, header=header_row - 1, dtype=str)
+    except Exception as e:
+        raise RuntimeError(f"Failed to read Excel file \"{source_file}\" or sheet \"{source_sheet}\":\n\t\t {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)